addons/l10n_my_edi/models/account_move.py

```diff
@@ -450,11 +451,16 @@ def _cron_l10n_my_edi_synchronize_myinvois(self):
         # A invoice whose status will not change anymore is: (cancelled or invalid) or has been validated more than 74h ago.
         # /!\ when an invoice validation is pending, l10n_my_edi_validation_time is still None. These also need to be updated.
         datetime_threshold = datetime.datetime.now() - datetime.timedelta(hours=74)
-        invoices = self.env['account.move'].search([
-            ('l10n_my_edi_state', 'not in', (False, 'invalid', 'cancelled')),  # In practice we could ignore rejected invoices as we don't fully support vendor bills.
-            '|', ('l10n_my_edi_validation_time', '>', datetime_threshold),
-            ('l10n_my_edi_validation_time', '=', None),
-        ])
+        # We always want to fetch in_progress invoices, it's very likely that their status is already there.
+        invoice_domain = [('l10n_my_edi_state', '=', 'in_progress')]
+        # For valid invoices, we want them if their l10n_my_edi_validation_time is less than 74h ago, and if their l10n_my_edi_retry_at in the past.
+        invoice_domain = expression.OR([invoice_domain, [
+            ('l10n_my_edi_state', '=', 'valid'),
+            ('l10n_my_edi_validation_time', '>', datetime_threshold),
+            ('l10n_my_edi_retry_at', '<=', datetime.datetime.now()),
+        ]])
+        invoices = self.env['account.move'].search(invoice_domain)
```

Comment:
Shouldn't we also add a limit and re-trigger the cron if we reached the limit?

We don't want to block other crons from running, especially with the introduced sleep

Reply:
We can't here, as you could have 100 moves belonging to the same batch and thus that would only trigger a single query.
There is a limit in python though, we process up to 25 batches at most